tests/generators/cpp/src/test.cpp

```diff
@@ -0,0 +1,353 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <string_view>
+#include <type_traits>
+
+#include <catch2/catch.hpp>
+
+#if defined OPENASSETIO_TRAITGENTEST_INCLUDES_PACKAGE
+#include <openassetio_traitgen_test_all/openassetio_traitgen_test_all.hpp>
+#elif defined OPENASSETIO_TRAITGENTEST_INCLUDES_SUBPACKAGE
+#include <openassetio_traitgen_test_all/specifications/specifications.hpp>
+#include <openassetio_traitgen_test_all/traits/traits.hpp>
+#elif defined OPENASSETIO_TRAITGENTEST_INCLUDES_NAMESPACE
+#include <openassetio_traitgen_test_all/specifications/test.hpp>
+#include <openassetio_traitgen_test_all/traits/aNamespace.hpp>
+#include <openassetio_traitgen_test_all/traits/anotherNamespace.hpp>
+#else
+#error "An #include style must be chosen"
+#endif
+
+TEST_CASE("openassetio_traitgen_test_all - all expected traits are defined") {
```

Comment:
[fix] Sorry if I missed it, but I think we need the equivalent for the Specifications too.

Reply:
Python is fine, just missed the C++ equivalent.  Added existence check in 7fb59657 then added `kTraitSet` check in ba318e6b

Reply:
Do we have anything equivalent to [this](https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/blob/6e136f622ca24255946b6b484f67e8672f18ca82/tests/generators/test_python.py#L336) too? That ensure we're not making empty containers.

Reply:
> Do we have anything equivalent to [this](https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/blob/6e136f622ca24255946b6b484f67e8672f18ca82/tests/generators/test_python.py#L336) too? That ensure we're not making empty containers.

Crafted something roughly equivalent (tricky in C++) in 5cae1752